src/core/shape_inference/include/scaled_dot_product_attention_shape_inference.hpp

```diff
@@ -73,9 +73,9 @@ std::vector<TRShape> shape_infer(const ScaledDotProductAttention* op,
         const auto& attention_mask_rank = attention_mask.rank();
         if (attention_mask_rank.is_static() && attention_mask_rank != 0) {
             const auto& attention_mask_rank_len = attention_mask_rank.get_length();
-            bool attention_mask_input_correctness = attention_mask_rank_len >= 2 &&
-                                                    DimType::merge(l_dim, l_dim, *(attention_mask.end() - 2)) &&
-                                                    DimType::merge(s_dim, s_dim, *(attention_mask.end() - 1));
+            bool attention_mask_input_correctness =
+                attention_mask_rank_len >= 2 && DimType::broadcast_merge(l_dim, l_dim, *(attention_mask.end() - 2)) &&
+                DimType::broadcast_merge(s_dim, s_dim, *(attention_mask.end() - 1));
```

Comment:
Would be great to cover this change with some tests, at least by "type_prop" for the op creation. 
https://github.com/openvinotoolkit/openvino/blob/eededc2a3a55a3c45a2e8053313d031bb35ce445/src/core/tests/type_prop/scaled_dot_product_attention.cpp#L250-L260
It should be also verified that plugins are able to broadcast the `attention_mask`, and cover such scenario in the plugins tests. Let's create a ticket for the tests if the fix is urgent.